tools/metatoenv.py

```diff
@@ -136,13 +152,31 @@ def _jinja_filter(dependencies, platform):
                                        "unmatched square brackets or closing bracket "
                                        "found before opening bracket: {}".format(key))
                 selector = key[left:right + 1]
-                if selector.startswith('[not'):
-                    if (platform in selector) or (selector.replace(
-                            '[not', '')[:-1].strip() in platform):
+                raw_selector = selector.lstrip('[ ').rstrip(' ]').replace(
+                    'python', 'py').replace('64', '').replace('32', '')
+
+                if raw_selector.startswith('py'):
+                    # Check for python version
+                    select_ver = raw_selector.replace('>',
+                                                      '=').replace('<',
+                                                                   '=').split('=')[-1]
+                    select_op = raw_selector.replace('py', '').replace(select_ver,
+                                                                       '').strip()
+                    if select_ver.count('.') == 0:
+                        select_ver = f'{select_ver[0]}.{select_ver[1:]}'
```

Comment:
This doesn't look right:
```.py
>>> s=['2']
>>> f'{s[0]}.{s[1:]}'
'2.[]'
```
Did you mean to test for `select_ver.count('.') != 0`?

Reply:
it was to convert `38` to `3.8`.
but i see your point that it doesn't work with a single character...